lib/puppet/defaults.rb

```diff
@@ -328,6 +332,29 @@ def self.default_diffargs
         for all subsequent attempts to load the feature.  This behavior is almost
         always appropriate for the server, and can result in a significant performance
         improvement for features that are checked frequently.
+
+        This setting is deprecated and has been replaced by always_cache_misses.
+      EOT
+    },
+    :always_cache_misses => {
```

Comment:
Ping @nfagerlund . Wanted to see if you had any ideas for a better name for this setting.